numpy/lib/function_base.py

```diff
@@ -3957,11 +3957,15 @@ def _quantile_is_valid(q):
     # avoid expensive reductions, relevant for arrays with < O(1000) elements
     if q.ndim == 1 and q.size < 10:
         for i in range(q.size):
-            if q[i] < 0.0 or q[i] > 1.0:
+            if q[i] < 0.0 or q[i] > 1.0 or np.isnan(q[i]):
```

Comment:
Calling `np.isnan` here will be very expensive (compared to the "optimization" of avoiding the `count_nonzero` call below.  Lets just rephrase this into `0.0 < q[i] < 1.0` and use the "positive" instead of "negative" definition also in the branch below.

The second huge advantage is that it avoid `np.isnan` which doesn't work well for object arrays (e.g. containing decimals), which is supported currently.

Reply:
That seems cleaner than the workarounds I used for `isnan` and `Fraction`